applications/StructuralMechanicsApplication/custom_elements/mass_element.cpp

```diff
@@ -0,0 +1,382 @@
+        rLeftHandSideMatrix.resize(system_size, system_size, false);
+    }
+
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(system_size, system_size);
+}
+
+void MassElement::CalculateRightHandSide(VectorType& rRightHandSideVector, const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+    const SizeType system_size = GetGeometry().PointsNumber() * 3;
+    if (rRightHandSideVector.size() != system_size) {
+        rRightHandSideVector.resize(system_size, false);
+    }
+
+    rRightHandSideVector = ZeroVector(system_size);
+
+    const auto& r_geom = GetGeometry();
+    const SizeType number_of_nodes = r_geom.PointsNumber();
+
+    Vector lump_fact = ZeroVector(number_of_nodes);
+    r_geom.LumpingFactors(lump_fact);
+
+    for (SizeType i=0; i<number_of_nodes; ++i) {
+        const double temp = lump_fact[i] * mMass;
+
+        for (SizeType j=0; j<3; ++j) {
+            const SizeType index = i * 3 + j;
+            rRightHandSideVector[index] += temp * r_geom[i].FastGetSolutionStepValue(VOLUME_ACCELERATION)[j];
+        }
+    }
+    KRATOS_CATCH("")
+}
+
+void MassElement::CalculateMassMatrix(MatrixType& rMassMatrix, const ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY
+    const auto& r_geom = GetGeometry();
+
+    // lumped mass matrix
+    const SizeType number_of_nodes = GetGeometry().PointsNumber();
+    const SizeType system_size = number_of_nodes * 3;
+
+    if (rMassMatrix.size1() != system_size) {
+        rMassMatrix.resize(system_size, system_size, false);
+    }
+
+    noalias(rMassMatrix) = ZeroMatrix(system_size, system_size);
+
+    Vector lump_fact = ZeroVector(number_of_nodes);
+    r_geom.LumpingFactors(lump_fact);
+
+    for (SizeType i=0; i<number_of_nodes; ++i) {
+        const double temp = lump_fact[i] * mMass;
+
+        for (SizeType j=0; j<3; ++j) {
+            const SizeType index = i * 3 + j;
+            rMassMatrix(index, index) = temp;
+        }
+    }
+
+    KRATOS_CATCH("")
+}
+
+void MassElement::CalculateDampingMatrix(MatrixType& rDampingMatrix, const ProcessInfo& rCurrentProcessInfo)
+{
+    StructuralMechanicsElementUtilities::CalculateRayleighDampingMatrix(
+        *this,
+        rDampingMatrix,
+        rCurrentProcessInfo,
+        GetGeometry().PointsNumber() * 3);
+}
+
+double MassElement::GetElementMass() const
+{
+    // TODO this should be coming from total_mass_process (currently not working bcs not threadsafe!)
+    // => we need a better solution for accessing things in the initial condiguration
+    KRATOS_TRY
+
+    const auto& r_geom = GetGeometry();
+    const SizeType local_dim = GetGeometry().LocalSpaceDimension();
+
+    double total_mass;
+
+    if (local_dim == 1) { // line
+        total_mass = GetProperties()[CROSS_AREA] * StructuralMechanicsElementUtilities::CalculateReferenceLength3D2N(*this);
+
+    } else if (local_dim == 2) { // tri / quad
+        total_mass = GetProperties()[THICKNESS]* r_geom.Area();
+
+    } else {

```

Comment:
Not 3D?

Reply:
Well volumetric

Reply:
I thought of it but I couldn't think of what that would represent in real life
do you have sth in mind?

In any case it can be easily extended if someone needs it / has an application